server/src/main/java/org/opensearch/cluster/routing/WeightedRoutingService.java

```diff
@@ -94,19 +98,21 @@ public ClusterState execute(ClusterState currentState) {
                 Metadata metadata = currentState.metadata();
                 Metadata.Builder mdBuilder = Metadata.builder(currentState.metadata());
                 WeightedRoutingMetadata weightedRoutingMetadata = metadata.custom(WeightedRoutingMetadata.TYPE);
+                ensureNoVersionConflict(requestVersion, weightedRoutingMetadata);
                 if (weightedRoutingMetadata == null) {
-                    logger.info("put weighted routing weights in metadata [{}]", request.getWeightedRouting());
-                    weightedRoutingMetadata = new WeightedRoutingMetadata(request.getWeightedRouting());
+                    logger.info("add weighted routing weights in metadata [{}]", newWeightedRouting);
+                    weightedRoutingMetadata = new WeightedRoutingMetadata(newWeightedRouting, requestVersion + 1);
                 } else {
-                    if (!checkIfSameWeightsInMetadata(newWeightedRoutingMetadata, weightedRoutingMetadata)) {
-                        logger.info("updated weighted routing weights [{}] in metadata", request.getWeightedRouting());
-                        weightedRoutingMetadata = new WeightedRoutingMetadata(newWeightedRoutingMetadata.getWeightedRouting());
+                    if (!checkIfSameWeightsInMetadata(newWeightedRouting, weightedRoutingMetadata.getWeightedRouting())) {
+                        logger.info("updated weighted routing weights [{}] in metadata", newWeightedRouting);
+                        weightedRoutingMetadata = new WeightedRoutingMetadata(newWeightedRouting, weightedRoutingMetadata.getVersion() + 1);
```

Comment:
nit : lets use `requestVersion + 1` instead of `weightedRoutingMetadata.getVersion() + 1` just to maintain consistency.